app/components/views/ocr/ocr.py

```diff
@@ -23,13 +23,41 @@
 from ..image import BaseImageView
 from .base import BaseOCRView
 from services import State
+from components.popups import TextDialog
+from os.path import join
+from utils.scripts import logText
+from PyQt5 import QtCore
 
 
 class OCRView(BaseImageView, BaseOCRView):
     def __init__(self, parent: QMainWindow, state: State = None):
         super().__init__(parent, state)
         self.loadSettings()
+        self.textdiag = TextDialog()
 
     @pyqtSlot()
     def rubberBandStopped(self):
         super().rubberBandStopped()
+        
+    def handleTextResult(self, result):
+        try:
+            self.canvasText.hide()
+            self.textdiag.setText(result)
+            self.textdiag.show()
```

Comment:
`handleTextResult` is a handler for text recognition. Add a different handler for text translation. We need different handlers so that when translation fails to execute, OCR will not be affected.